src/test/scala/chiselTests/properties/PropertySpec.scala

```diff
@@ -617,4 +618,124 @@ class PropertySpec extends ChiselFlatSpec with MatchesAndOmits {
       lit.isLit shouldBe true
     })
   }
+
+  behavior.of("PropertyArithmeticOps")
+
+  it should "support expressions in temporaries, wires, and ports" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(new RawModule {
+      val a = IO(Input(Property[Int]()))
+      val b = IO(Input(Property[Int]()))
+      val c = IO(Output(Property[Int]()))
+      val d = IO(Output(Property[Int]()))
+      val e = IO(Output(Property[Int]()))
+
+      val t = a + b
+
+      val w = WireInit(t)
+
+      c := t
+      d := t + a
+      e := w + (a + b)
+    })
+
+    matchesAndOmits(chirrtl)(
+      "wire t : Integer",
+      "propassign t, integer_add(a, b)",
+      "wire w : Integer",
+      "propassign w, t",
+      "propassign c, t",
+      "wire _d_WIRE : Integer",
```

Comment:
Do you want to give a less jarring name than `_WIRE`? You could do `.autoSeed` on the wires you're creating and give them something descriptive, eg. `_propOp` or something (do start it with an `_` though so that it's not treated as an important name downstream). eg. see what I did here: https://github.com/chipsalliance/chisel/blob/ff3766b6741fc519d6e7dd86127c9ee3f1b43809/core/src/main/scala/chisel3/Bits.scala#L692.

Note that it isn't picking up your val name `_wire` because the Chisel compiler plugin is not run on the core of Chisel.